hbase-server/src/main/java/org/apache/hadoop/hbase/ipc/RpcServer.java

```diff
@@ -313,6 +314,7 @@ public void onConfigurationChange(Configuration newConf) {
     if (scheduler instanceof ConfigurationObserver) {
       ((ConfigurationObserver) scheduler).onConfigurationChange(newConf);
     }
+    HBasePolicyProvider.init(newConf, authManager);
```

Comment:
Regarding thread-safety, do we need to synchronize on authManager while this happens? Otherwise authorize() requests and this init can race and can result in some weird authz results..

Reply:
Fixed